tests/pos-with-compiler-cc/dotc/util/SimpleIdentitySet.scala

```diff
@@ -7,7 +7,7 @@ import collection.mutable
 /** A simple linked set with `eq` as the comparison, optimized for small sets.
  *  It has linear complexity for `contains`, `+`, and `-`.
  */
-abstract class SimpleIdentitySet[+Elem <: AnyRef] {
+sealed abstract class SimpleIdentitySet[+Elem <: AnyRef] {
```

Comment:
What breaks without `sealed`?

Reply:
The inferred self type of `SimpleIdentitySet`. With `sealed` we can compute it to be pure by looking at the subclasses. 

We could also have declared the self type as pure - either directly or indirectly by inheriting from `caps.Pure` - but I wanted to experiment with `sealed` to check that it works correctly